src/js/reducers.js

```diff
@@ -506,6 +508,13 @@ function ui(state = {}, action) {
             var newState = { ...state }
             var app = newState[action.appID] ? newState[action.appID] : newAppState()
             return newState
+        case Actions.SET_GLOBAL_PROPERTIES:
+            var newState = { ...state }
+            var app = newState[action.appID] ? newState[action.appID] : newAppState()
+            if (action.menuLayout && action.menuLayout.length) {
+                app.menuLayout = action.menuLayout
+            }
```

Comment:
I know this is not done everywhere, but i think this is better practice to include this line before returning newState.
```
newState[action.appID] = app
```

Reply:
added in https://github.com/smartdevicelink/generic_hmi/pull/179/commits/a51b2fb95d7f7620d01a9566135817583345a4d1